go/vt/vtgate/planbuilder/from.go

```diff
@@ -43,7 +43,9 @@ func processTableExpr(tableExpr sqlparser.TableExpr, vschema VSchema) (builder,
 		// more routes can be merged with this one. If so, the order
 		// should be maintained as dictated by the parenthesis.
 		if rb, ok := bldr.(*route); ok {
-			rb.Select.From = sqlparser.TableExprs{tableExpr}
+			if sel, ok := rb.Select.(*sqlparser.Select); ok {
```

Comment:
This is correct. However, I'd go one step further and assertively expect this to be a Select. So, go with this instead: `rb.Select.(*sqlparser.Select).From = ...`.
Add a comment explaining why: "If a route is returned in this context, then we know that it only contains the from clause of a brand new, partially built Select. If there was a subquery, it would have been aliased to a name and the entire thing would still be the from clause of a partially built Select."